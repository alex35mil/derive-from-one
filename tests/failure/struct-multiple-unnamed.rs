use derive_from_one::FromOne;

#[derive(FromOne)]
pub struct T(u32, bool);

fn main() {
    let _: T = T::from(1);
}
