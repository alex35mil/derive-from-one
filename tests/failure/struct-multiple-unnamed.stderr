error: FromOne can only be used with structs with a single field
 --> tests/failure/struct-multiple-unnamed.rs:4:12
  |
4 | pub struct T(u32, bool);
  |            ^

error[E0308]: mismatched types
 --> tests/failure/struct-multiple-unnamed.rs:7:24
  |
7 |     let _: T = T::from(1);
  |                ------- ^ expected `T`, found integer
  |                |
  |                arguments to this function are incorrect
  |
note: associated function defined here
 --> $RUST/core/src/convert/mod.rs
  |
  |     fn from(value: T) -> Self;
  |        ^^^^
